File: ELEC881_Copilot/src/No Description Prompt/Python/q30-frog-jump.py
class Solution(object):
    def canCross(self, stones):
        """
        :type stones: List[int]
        :rtype: bool
        """
        if stones[1] != 1:
            return False
        n = len(stones)
        dp = [set() for _ in range(n)]
        dp[1].add(1)
        for i in range(2, n):
            for j in range(i):
                k = stones[i] - stones[j]
                if k in dp[j]:
                    dp[i].add(k)
                if k - 1 in dp[j]:
                    dp[i].add(k - 1)
                if k + 1 in dp[j]:
                    dp[i].add(k + 1)
        return len(dp[-1]) > 0

File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q30-frog-jump.js
/**
 * @param {number[]} stones
 * @return {boolean}
 */
var canCross = function(stones) {
    const n = stones.length;
    const dp = Array(n).fill(0).map(() => Array(n).fill(false));
    dp[0][0] = true;
    for (let i = 1; i < n; i++) {
        if (stones[i] - stones[i - 1] > i) {
            return false;
        }
    }
    for (let i = 1; i < n; i++) {
        for (let j = i - 1; j >= 0; j--) {
            const k = stones[i] - stones[j];
            if (k > j + 1) {
                break;
            }
            dp[i][k] = dp[j][k - 1] || dp[j][k] || dp[j][k + 1];
            if (i === n - 1 && dp[i][k]) {
                return true;
            }
        }
    }
    return false;
    
};